### PR TITLE
[MOB-129] PUT Activate account

### PR DESCRIPTION
**Pull request by JM-Abrodos**
## Pull Request Checklist

- [x] **Title**
- [x] **[MOB US' Link](https://mobilemakers.atlassian.net/browse/MOB-129)**
- [ ] **Link PR to US**
- [x] **Add Tests**
- [ ] **Add ENV Variables and Validations**
- [ ] **Documentation Updates**
- [x] **Swagger Updates**

## Testing Guide
Example curl:
```
curl --location --request PUT 'http://localhost:3000/v1/account/activate' \
--header 'Content-Type: application/json' \
--data '{
    "verificationCode": "942428"
}'
```

### CODE REVIEW

File: src/infrastructure/http/controllers/account.controller.ts [hunk 1]
```diff
@@ -196,6 +201,37 @@ export class AccountController {
...
+  @Put('/activate')
+  @SweggerActivateAccount()
+  @HttpCode(200)
+  async activate(
+    @Body() activateAccountDto: ActivateAccountDto,
+    @RequestId() requestId: string,
+  ): Promise<AccountLoggedDto> {
+    try {
+      this.logger.info(
+        `[AccountController] [activateAccount] - x-request-id:${requestId}`,
```
Comment (LuciaAccenture) [inline src/infrastructure/http/controllers/account.controller.ts:213]: It's silly but we are putting the name of the function as it is in all endpoints, it should be `[AccountController] [activate]`

File: src/infrastructure/db/repositories/account.repository.ts [hunk 2]
```diff
@@ -117,6 +118,47 @@ export class AccountRepository extends MongoRepository<UserSchema> {
...
   }
 
+  /**
+   * Update status to ACTIVE and return the document updated.
+   *
+   * @param email - User email address.
+   * @param requestId - Request identifier.
+   * @returns UserProfileDto - User data following business schema.
+   */
+  async acticateAccountByEmail(
```
Comment (LuciaAccenture) [inline src/infrastructure/db/repositories/account.repository.ts:128]: Be careful, here you have made a mistake when writing the function `activateAccountByEmail`

File: src/infrastructure/db/repositories/account.repository.ts [hunk 3]
```diff
@@ -117,6 +118,47 @@ export class AccountRepository extends MongoRepository<UserSchema> {
...
+    email: string,
+    requestId?: string,
+  ): Promise<UserProfileDto> {
+    try {
+      this.logger.info(
+        `[AccountRepository] [acticateAccountByEmail] - x-request-id: ${requestId}, email: ${email}`,
```
Comment (LuciaAccenture) [inline src/infrastructure/db/repositories/account.repository.ts:134]: Change the name of the function in the logs also `acticateAccountByEmail`

File: src/infrastructure/db/repositories/mongo.repository.ts [hunk 4]
```diff
@@ -145,6 +169,20 @@ export class MongoRepository<E> implements DBRepository<E> {
     );
   }
 
+  /**
+   * Finde a document by email, update the status and return the document updated.
+   *
+   * @param email - User email address.
+   * @param newStatus - Status value to update.
+   * @returns
```
Comment (LuciaAccenture) [inline src/infrastructure/db/repositories/mongo.repository.ts:177]: You forgot to complete this comment.

File: src/infrastructure/http/controllers/account.controller.ts [hunk 5]
```diff
@@ -35,6 +38,7 @@ import {
   SwaggerDisjoinEvent,
   SwaggerRefreshToken,
   SwaggetGetMyEvents,
+  SweggerActivateAccount,
```
Comment (LuciaAccenture) [inline src/infrastructure/http/controllers/account.controller.ts:41]: You've made a mistake, change the name to ` SwaggerActivateAccount`

File: test/mocks/db/temporal-code.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { TEMPORAL_CODE_TYPE } from '../../../src/domain/const';
+import { EMAIL_MOCK_DB } from './user';
+
+export const VERIFICATION_CODE_DATA_MOCK = {
```
Comment (LuciaAccenture) [inline test/mocks/db/temporal-code.ts:4]: Following the naming of what we are using in the rest of files this should be `VERIFICATION_CODE_DATA_MOCK_DB`

File: test/mocks/db/user.ts [hunk 7]
```diff
@@ -1,16 +1,19 @@
...
 export const PASSWORD_MOCK_DB = 'Password3';
 
+export const EMAIL_MOCK_DB = 'esmorga.test.03@yopmail.com';
+
 const USER_MOCK_DB = {
   _id: USER_MOCK_DB_ID,
   name: 'Scottie',
   lastName: 'Pippen',
-  email: 'esmorga.test.03@yopmail.com',
+  email: EMAIL_MOCK_DB,
```
Comment (LuciaAccenture) [inline test/mocks/db/user.ts:13]: Here I have a doubt, in the `ADMIN_USER_MOCK_DB` shouldn't you also put the `_id` and `email` with a mock?`

Comment (JM-Abrodos) [reply test/mocks/db/user.ts:13]: That properties used for normal user are imported in other mocks. Admin account is used in specific flows only

**Review by LuciaAccenture — APPROVED** [review]

**Pull request merged**